react/src/pages/AccountsComponent.js

```diff
@@ -19,390 +44,329 @@ function isCurrentAccount(header, info) {
+            return <>Expires {Time.FromNow(portalAccessKeyResponse.expires_at)} {Char.RightArrow} {portalAccessKeyResponse.expires_at}</>
+        }
+    }
+    else {
+        return <>No expiration</>
+    }
+}
 
-    const SslCertificateLink = ({ url }) => {
-        return url && url.startsWith("https://") && <small>
-          &nbsp;| <a style={{color:"inherit"}} href={Client.Path("certificates") + "/?hostname=" + url} rel="noreferrer" target="_blank"><b>SSL</b></a>&nbsp;
-          <a style={{color:"inherit"}} href={Client.Path("certificates") + "/?hostname=" + url} rel="noreferrer" target="_blank"><span className="fa fa-external-link" style={{position:"relative",bottom:"-1px"}}></span></a>
-        </small>
+const SslCertificateLink = ({ url }) => {
+    const sslCertificate = useFetch(url ? `/certificates?hostname=${url}` : null);
+    function sslCertificateExpiresSoon() {
+        return Type.IsArray(sslCertificate.data) ? sslCertificate.data[0]?.expires_soon : null;
+    }
+    function sslCertificateExpiresMessage() {
+        if (!Type.IsArray(sslCertificate.data)) return "";
+        const invalid = sslCertificate.data[0]?.invalid;
+        const expired = sslCertificate.data[0]?.expired;
+        const expiresSoon = sslCertificate.data[0]?.expired;
+        const expiresAt = sslCertificate.data[0]?.expires_at;
+        if (invalid) {
+            return `SSL certificate is invalid!`;
+        }
+        else if (expired) {
+            return `SSL certificate expired ${Time.Ago(expiresAt)} (${expiresAt})`;
+        }
+        else if (expiresSoon) {
+            return `SSL certificate expires ${Time.FromNow(expiresAt)} (${expiresAt}) -> Soon!`;
+        }
+        else {
+            return `SSL certificate expires ${Time.FromNow(expiresAt)} (${expiresAt})`;
+        }
     }
+    const uuid = Uuid();
+    return url && url.startsWith("https://") && <small>
+        <small style={{marginLeft:"3pt",marginRight:"3pt"}}>|</small>
+        <a style={{color:sslCertificateExpiresSoon() ? "red" : "inherit"}} href={Client.Path("certificates") + "/?hostname=" + url} id={`tooltip-ssl-expires-${url}`} rel="noreferrer" target="_blank">SSL</a>&nbsp;
+        <ExternalLink
+            href={Client.Path("certificates") + "/?hostname=" + url}
+            style={{marginLeft:"1pt"}} />
+        <Tooltip id={`tooltip-ssl-expires-${url}`} text={`${sslCertificateExpiresMessage()}`} position="top" />
+    </small>
+}
 
-    const portalAccessKey = useFetch("/portal_access_key");
+const S3BucketLink = ({ bucket, name, line = true }) => {
+    if (!bucket) return <></>
+    return <>
+        { name && <>
+            <b>{name}</b>:&nbsp;&nbsp;
+        </> }
+        <a href={`https://s3.console.aws.amazon.com/s3/buckets/${bucket}`} style={{color:"inherit"}} rel="noreferrer" target="_blank">{bucket}</a>
+        <ExternalLink
+            href={`https://s3.console.aws.amazon.com/s3/buckets/${bucket}`}
+            style={{marginLeft:"4pt"}} />
```

Comment:
Does this making the 4pt inside the underscoring or other link decoration? If not, why its this spacing part of the link and not a thin space (`&thinsp;`) or something? Just curious.